tools/azsdk-cli/docs/specs/typespec-authoring.spec.md

```diff
@@ -0,0 +1,518 @@
+- New team members face an especially steep learning curve without AI assistance that understands Azure patterns
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+What are we trying to achieve with this design?
+
+- [ ] **AI Pair Programming for TypeSpec**: Enable GitHub Copilot to provide intelligent, context-aware assistance for TypeSpec authoring by integrating Azure SDK RAG (Retrieval-Augmented Generation) knowledge base
+- [ ] **Guide Users Through Intent-Driven Development**: Allow users to describe their intent in natural language (e.g., "I need to add a new API version to my Widget service" or "I want to add an ARM resource named 'Asset' with CRUD operations"), and have the AI guide them through the correct TypeSpec implementation
+- [ ] **Generate Code Following ARM/DP/SDK/TypeSpec Guidelines**: Ensure that generated TypeSpec code adheres to Azure Resource Manager (ARM) patterns, Data Plane (DP) standards, SDK guidelines, and TypeSpec best practices
+- [ ] **Provide Contextual References**: When generating or suggesting code, include references to relevant documentation (e.g., links to TypeSpec Azure guidelines for versioning, ARM resource types, routing patterns)
+- [ ] **Save Review Efforts**: Reduce the time reviewers spend identifying standards violations by ensuring code follows standards from the start
+- [ ] **Improve Developer Learning**: Help service teams learn TypeSpec syntax and Azure patterns through interactive guidance, increasing their confidence in making code changes
+- [ ] **Accelerate Inner Loop Development**: Speed up the iterative process of authoring TypeSpec, compiling, validating, and adjusting to achieve expected SDK outputs
+
+### Exceptions and Limitations
+
+_Known cases where this approach doesn't work or has limitations._
+
+#### Language-Specific Limitations
+
+| Language   | Limitation | Impact | Workaround |
+|------------|------------|--------|------------|
+| .NET       | None (TypeSpec is language-agnostic at authoring time) | N/A | N/A |
+| Java       | None (TypeSpec is language-agnostic at authoring time) | N/A | N/A |
+| JavaScript | None (TypeSpec is language-agnostic at authoring time) | N/A | N/A |
+| Python     | None (TypeSpec is language-agnostic at authoring time) | N/A | N/A |
+| Go         | None (TypeSpec is language-agnostic at authoring time) | N/A | N/A |
+
+**Note:** TypeSpec authoring is language-agnostic. The generated SDKs target specific languages, but the TypeSpec authoring experience with AI assistance applies uniformly across all target SDK languages. Language-specific considerations come into play during SDK generation validation, not during TypeSpec authoring.
+
+---
+
+## Design Proposal
+
+Leverage the existing Azure SDK RAG service to deliver solutions aligned with Azure guidelines and best practices. Implement a TypeSpec authoring MCP tool that consults the Azure SDK RAG service to generate these solutions.
```

Comment:
We don't have the product of 'RAG service',  please use 'APIs for Azure SDK knowledge base'

Reply:
same for other places in this spec.